study1/week4/flood-fill.java

```diff
@@ -0,0 +1,30 @@
+// https://leetcode.com/problems/flood-fill/
+// Time Complexity: O(n*m), n: image.length, m: image[0].length
+// Space Complexity: O(1)
+// using recursive methods
+class Solution {
+    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
+        if (image[sr][sc] == color) {
+            return image;
+        }
+
+        int pixelColor = image[sr][sc]; // original color
+
+        image[sr][sc] = color; // change the color of the pixel
+
+        int[][] dir = new int[][]{{0, 1},{0, -1},{1, 0},{-1, 0}}; // up, down, right, left
```

Comment:
This variable should be rather `static`  to `class Solution`, because this function is called multiple times and also being assigned multiple times, if you make it static you will be able to safe multiple operations and space.